### PR TITLE
Fix for issues in cookie file.

### PR DESCRIPTION
**Pull request by lucifercr07**
- cookie.js:
Issue - PropertyList should be required module for cookie.js file
Fix: Added PropertyList as required module.

Added unit-tests corresponding to the changes.

### CODE REVIEW

File: test/unit/request-auth.test.js [hunk 1]
```diff
@@ -123,4 +123,29 @@ describe('RequestAuth', function () {
             fakeParent: true
         });
     });
+
+    describe('parameters', function () {
```
Comment (coditva) [inline test/unit/request-auth.test.js:127]: I think we're testing `.update` here.

File: test/unit/request-auth.test.js [hunk 2]
```diff
@@ -123,4 +123,29 @@ describe('RequestAuth', function () {
...
+        it('should return with already set auth params if type set to invalid auth type', function () {
```
Comment (coditva) [inline test/unit/request-auth.test.js:128]:
```suggestion
	describe('.update', function () {
        it('should not update auth params if type is invalid', function () {
```

**Review by coditva — CHANGES_REQUESTED** [review]
Added some comments

File: test/unit/request-auth.test.js (hunk 2 above)
Comment (lucifercr07) [reply test/unit/request-auth.test.js:128]: Moving request-auth changes to separate PR as suggested by @codenirvana https://github.com/postmanlabs/postman-collection/pull/1101

File: test/unit/cookie.test.js [hunk 3]
```diff
@@ -194,4 +194,26 @@ describe('Cookie', function () {
             })).valueOf()).to.eql('this is a cookie value');
         });
     });
+
+    describe('unparse', function () {
+        it('should return empty string if not array of cookies', function () {
```
Comment (coditva) [inline test/unit/cookie.test.js:199]: The `it` block name does not specify what exactly is being tested here.
```suggestion
        it('should return empty string if called with invalid cookies array', function () {
```

**Review by coditva — CHANGES_REQUESTED** [review]

**Review by coditva — APPROVED** [review]

**Comment by coditva** [comment]
LGTM! 🚀

File: CHANGELOG.yaml [hunk 4]
```diff
@@ -1,3 +1,9 @@
+master:
+  fixed bugs:
```
Comment (codenirvana) [inline CHANGELOG.yaml:2]: Then why's this PR fixing a bug where `Cookie. toString` is not serializing the `extensions` properly?

File: lib/collection/cookie.js [hunk 5]
```diff
@@ -270,7 +271,7 @@ _.assign(Cookie.prototype, /** @lends Cookie.prototype */ {
 
         if (this.extensions) {
             this.extensions.forEach(function (ext) {
-                str += '; ' + ext;
+                str += '; ' + ext.key + '=' + ext.value;
```
[the remark this answers is not in the document]
Comment (codenirvana) [reply lib/collection/cookie.js:274]: Yes, separate PR.

File: test/unit/cookie.test.js [hunk 6]
```diff
@@ -194,4 +194,26 @@ describe('Cookie', function () {
             })).valueOf()).to.eql('this is a cookie value');
         });
     });
+
+    describe('unparse', function () {
+        it('should return empty string if called with invalid cookies array', function () {
```
Comment (codenirvana) [inline test/unit/cookie.test.js:199]:
```suggestion
        it('should return empty string on invalid input types', function () {
```

File: test/unit/cookie.test.js [hunk 7]
```diff
@@ -194,4 +194,26 @@ describe('Cookie', function () {
...
+            expect(Cookie.unparse('')).to.equal('');
```
Comment (codenirvana) [inline test/unit/cookie.test.js:200]: can we add assertions for other invalid types?

**Review by codenirvana — CHANGES_REQUESTED** [review]

File: CHANGELOG.yaml (hunk 4 above)
Comment (lucifercr07) [reply CHANGELOG.yaml:2]: Removed the `extensions` related part, will raise as part of separate PR.

File: lib/collection/cookie.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (lucifercr07) [reply lib/collection/cookie.js:274]: Will raise as separate PR.

File: test/unit/cookie.test.js (hunk 6 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

File: test/unit/cookie.test.js (hunk 7 above)
Comment (lucifercr07) [reply test/unit/cookie.test.js]: Fixed.

**Review by codenirvana — DISMISSED** [review]

**Review by codenirvana — DISMISSED** [review]

**Review by codenirvana — APPROVED** [review]

**Pull request merged**